### DISCUSSION TITLE
Tiled Diffusion for ComfyUI Tutorial

### DISCUSSION THREAD

**Discussion by JPGranizo** [shiimizu/ComfyUI-TiledDiffusion#8] [Q&A] [answered]
Hi! Thank you so much for migrating Tiled diffusion / Multidiffusion and Tiled VAE to ComfyUI. Because of that I am migrating my workflows from A1111 to Comfy. 

As a beginner, it is a bit difficult, however, to set up Tiled Diffusion plus ControlNet Tile upscaling from scatch. I improted you png Example Workflows, but I cannot reproduce the results. 

Therefore, I would be immensely grateful if you could possibly provide a bit of a more elaborate tutorial which allows to exactly reproduce the A1111 settings and workflows.

Thank you!

**Comment by shiimizu** [comment, marked as answer]
Hi! Welcome aboard on the noodle train!

If you're starting out in ComfyUI, I can point you to some resources:
* Informative [video tutorials](https://www.youtube.com/watch?v=_C7kR2TFIX0) by the developer at [ComfyUI_IPAdapter_plus](https://github.com/cubiq/ComfyUI_IPAdapter_plus).  
* [Official examples](https://comfyanonymous.github.io/ComfyUI_examples/), notably the [“Hires Fix” aka 2 Pass Txt2Img](https://comfyanonymous.github.io/ComfyUI_examples/2_pass_txt2img) and [ControlNets and T2I-Adapter](https://comfyanonymous.github.io/ComfyUI_examples/controlnet) page.
* [Keyboard shortcuts](https://github.com/comfyanonymous/ComfyUI#shortcuts)

While it'd be difficult to *exactly* reproduce the A1111 settings and workflows, we can definitely try.  
Let's try recreating an image from CivitAI. We're going to use my [nodes](https://github.com/shiimizu/ComfyUI_smZNodes) to try to recreate A1111 gens. Let's work with [this image](https://civitai.com/images/506097). Drag and drop that in ComfyUI to load it. That alone will get us ~90% there.

At this point, here's what I did to the workflow:
1. Double check the checkpoint, vae, and upscaler. (As well as LoRas, if there are any)
2. In the `Settings` node, I see that noise will be generated on the GPU.
3. Connect Tiled Diffusion, Tiled VAE Encode/Decode, and ControlNet nodes in the hires pass.

To upscale further, clone all the nodes involved in the "hires" pass, and connect them to each other.
So it becomes like: txt2img pass -> hires pass -> hires pass. 

As for ControlNets, you'll need [ComfyUI-Advanced-ControlNet](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet) to get the same behaviour as "[My prompt is more important](https://github.com/Kosinkadink/ComfyUI-Advanced-ControlNet/issues/24#issuecomment-1789927761)", etc.

>[!NOTE]  
> * The original A1111 extension has their sizes set in latent-space, as shown [here](https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111/#tiled-diffusion), whenever it says `Latent tile` and then width/height/overlap/batch size. That was quite confusing to me starting out  since we normally work in pixel-space. All you have to do is multiply their values by 8 to get the pixel-space equivalent that you can enter in the `Tiled Diffusion` and `Tiled VAE Decode` node.
> * The Tiled VAE encode/decode nodes set the recommended `tile_size` upon creation. Those values are for my machine, so you may need to recreate those nodes. I only use those nodes if I'm generating a large image, of course that depends on your GPU.

And the final result that you can also load in ComfyUI (it looks different because we used ControlNet + Tiled Diffusion):

![ComfyUI_07543_](https://github.com/shiimizu/ComfyUI-TiledDiffusion/assets/54494639/a313a03d-d3f4-4cb3-88a4-ff38e4f2f65d)

As you can see, there's quite a bit involved in trying to reproduce A1111 gens, but with that comes more freedom and flexibility in the generation process. I hope that helps!

**Comment by JPGranizo** [comment]
@shiimizu, I want to express my deepest gratitude for the comprehensive and detailed explanation you provided. It has been invaluable in setting up this workflow

Additionally, the resources you linked for diving into comfyUI have been a great help to get started. I've made significant progress, though there's still plenty to explore.

Your support and assistance have been instrumental, and I can't thank you enough for developing such incredible comfyUI nodes. Your contributions are truly remarkable, and I encourage you to continue the excellent work!

**Comment by JPGranizo** [comment]
@shiimizu Thank you once again for your fantastic work on making Tiled Diffusion available for ComfyUI. It’s definitely a significant improvement over SDUltimateUpscaler, and it’s great to see that you’ve added support for Flux and SDXL.

Since your last update in this thread, have you made any interesting observations regarding settings for the newer models, like Flux? Are there any notable differences in settings? Have you identified optimal values for parameters like method, tile size, overlap, denoising, or any other key factors?
